ratis-netty/src/main/java/org/apache/ratis/netty/client/NettyClientStreamRpc.java

```diff
@@ -455,26 +454,23 @@ public CompletableFuture<DataStreamReply> streamAsync(DataStreamRequest request)
           flushRequestCountMin, flushRequestBytesMin, request)? channel::writeAndFlush: channel::write;
       channelFuture = writeMethod.apply(request);
     }
+
+    final TimeDuration timeout = isClose ? closeTimeout : requestTimeout;
+    replyEntry.scheduleTimeout(() -> channel.eventLoop().schedule(() -> {
```

Comment:
I think about this part of the code like this:
﻿
After the client sends successfully, then wait for the server's return, the timeout time is to detect the timeout of the server's return, so I start the timeout listener after the client sends the request successfully. So I started the timeout listener after the client sent the request successfully, and canceled the timeout listener after the server returned successfully.

Reply:
I see.  Then, let's keep the previous logic.